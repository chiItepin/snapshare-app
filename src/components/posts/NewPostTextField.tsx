import React, { FunctionComponent } from 'react';
import {
  View,
} from 'react-native';
import {
  Modal,
  Button,
  TextField,
} from 'react-native-ui-lib';
import styles from '../../styles/GlobalStyles';
import HelperStyles from '../../styles/HelperStyles';

interface IProps {
  isVisible: boolean;
  setIsNewPostTextFieldVisible: React.Dispatch<React.SetStateAction<boolean>>;
  value: string;
  onChange: React.Dispatch<React.SetStateAction<string>>;
  onSubmit: () => void;
}

const NewPostTextField: FunctionComponent<IProps> = ({
  isVisible,
  setIsNewPostTextFieldVisible,
  value,
  onChange,
  onSubmit,
}: IProps) => (
  <Modal visible={isVisible} animationType="slide">
    <View style={[styles.container, HelperStyles.row]}>
      <View style={[HelperStyles['w-50'], HelperStyles.paddingRightMed]}>
        <Button
          onPress={() => setIsNewPostTextFieldVisible(false)}
          label="Close"
          outline
          borderRadius={6}
        />
      </View>

      <View style={[HelperStyles['w-50'], HelperStyles.paddingLeftMed]}>
        <Button
          onPress={onSubmit}
          label="Submit"
          enableShadow
          borderRadius={6}
          disabled={value.length === 0}
        />
      </View>

      <View style={[HelperStyles['w-100'], HelperStyles.marginBottomBig]}>
        <TextField
          multiline
          placeholder="Enter content"
          value={value}
          onChangeText={onChange}
          showCharacterCounter
          maxLength={300}
        />
      </View>
    </View>
  </Modal>
);

export default NewPostTextField;
