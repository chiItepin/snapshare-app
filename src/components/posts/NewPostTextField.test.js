import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import NewPostTextField from './NewPostTextField';

test('NewPostTextField renders and handles events', () => {
  const mockFn = jest.fn();
  const props = {
    isVisible: true,
    value: 'lorem',
    onSubmit: mockFn,
    setIsNewPostTextFieldVisible: mockFn,
    onChange: mockFn,
    setNotificationMessage: mockFn,
    setImages: mockFn,
    images: [],
  };

  const component = render(
    <NewPostTextField {...props} />,
  );

  const textArea = component.getByDisplayValue(props.value);
  fireEvent.changeText(textArea, 'ipsum');

  const submitButton = component.getByText('Submit');
  fireEvent.press(submitButton);

  const cancelButton = component.getByText('Close');
  fireEvent.press(cancelButton);

  // 3 events triggered | 2 buttons and 1 field change triggered
  expect(mockFn).toHaveBeenCalledTimes(3);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
