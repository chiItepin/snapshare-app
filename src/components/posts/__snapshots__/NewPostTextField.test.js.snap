// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPostTextField renders and handles events 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  modifiers={
    Object {
      "alignments": Object {},
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "color": undefined,
      "flexStyle": undefined,
      "margins": Object {},
      "paddings": Object {},
      "positionStyle": undefined,
      "typography": undefined,
    }
  }
  visible={true}
>
  <View
    blurType="light"
    style={
      Array [
        undefined,
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "flex-start",
            "padding": 12,
          },
          Object {
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "50%",
            },
            Object {
              "paddingRight": 12,
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#5A48F5",
              "borderRadius": 6,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "minWidth": 90,
              "opacity": 1,
              "paddingHorizontal": 19,
              "paddingVertical": 8.5,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 0,
                    "flexDirection": "row",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#5A48F5",
                  },
                  Object {},
                  Object {},
                  undefined,
                ],
              ]
            }
          >
            Close
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "width": "50%",
            },
            Object {
              "paddingLeft": 12,
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5A48F5",
              "borderRadius": 6,
              "elevation": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "minWidth": 90,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 9.5,
              "shadowColor": "#5A48F5",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.35,
              "shadowRadius": 9.5,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 0,
                    "flexDirection": "row",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {},
                  Object {},
                  undefined,
                ],
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
            Object {
              "marginBottom": 25,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Array [
              undefined,
              undefined,
              undefined,
              undefined,
              Object {},
              Object {},
              Object {},
              Array [
                Object {},
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                Object {},
                Object {},
                Array [
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "#F0F2F5",
                    "flexDirection": "row",
                    "justifyContent": undefined,
                    "paddingBottom": 10,
                  },
                  undefined,
                  Object {
                    "borderColor": "#D2D6D8",
                  },
                  Object {
                    "paddingTop": undefined,
                  },
                ],
              ]
            }
          >
            <TextInput
              accessibilityLabel=""
              allowFontScaling={true}
              editable={true}
              enableErrors={true}
              maxLength={300}
              multiline={true}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter content"
              placeholderTextColor="#6E7881"
              rejectResponderTermination={true}
              showCharacterCounter={true}
              style={
                Array [
                  undefined,
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "transparent",
                    "borderWidth": undefined,
                    "flexGrow": 1,
                    "padding": 0,
                    "textAlign": "left",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  undefined,
                  false,
                  Object {
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#20303C",
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              validateOnBlur={true}
              value="ipsum"
            />
          </View>
          <View
            row={true}
            style={
              Array [
                undefined,
                undefined,
                undefined,
                undefined,
                Object {},
                Object {},
                Object {
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  Object {
                    "flex": 1,
                  },
                  undefined,
                  Object {},
                  Object {},
                  Object {},
                  undefined,
                ]
              }
            >
              <Text
                accessible={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#FC3D2F",
                        "textAlign": "left",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "height": 20,
                        "lineHeight": 20,
                      },
                      Object {
                        "marginTop": 9,
                      },
                      undefined,
                    ],
                  ]
                }
              />
            </View>
            <Text
              accessibilityLabel="5 out of 300 max characters"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "left",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#6E7881",
                    },
                    Object {
                      "marginTop": 9,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "height": 20,
                      "lineHeight": 20,
                    },
                  ],
                ]
              }
            >
              5
               / 
              300
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
